stan/math/fwd/core/std_complex.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef STAN_MATH_FWD_CORE_STD_COMPLEX_HPP
+#define STAN_MATH_FWD_CORE_STD_COMPLEX_HPP
+
+#include <stan/math/prim/core/complex_base.hpp>
+#include <stan/math/fwd/core/fvar.hpp>
+#include <complex>
+#include <type_traits>
+
+namespace std {
+
+/**
+ * Specialization of the standard libary complex number type for
+ * reverse-mode autodiff type `stan::math::fvar<T>`.
+ */
+template <typename T>
+class complex<stan::math::fvar<T>>
+    : public stan::math::complex_base<stan::math::fvar<T>> {
+ public:
+  using base_t = stan::math::complex_base<stan::math::fvar<T>>;
+  using value_type = stan::math::fvar<T>;
+  using complex_type = complex<value_type>;
+
+  /**
+   * Construct complex number from real and imaginary parts.
+   *
+   * @tparam U type of real part
+   * @tparam V type of imaginary part
+   * @param[in] re real part
+   * @param[in] im imaginary part
+   */
+  template <typename U, typename V>
+  complex(const U& re, const V& im) : base_t(re, im) {}
+
+  /**
+   * Constructs complex number from argument, which may be a scalar or
+   * a complex number.  If it is a scalar, the real part is set to the
+   * scalar value and the imaginary part to zero;  if it is complex,
+   * the real and imaginary parts are set to those of the argument.
+   *
+   * @tparam U type of argument
+   * @param[in] x the scalar or complex argument
+   */
+  template <typename U>
+  complex(const U& x) : base_t(x) {}  // NOLINT(runtime/explicit)
+
+  /**
+   * Construct a complex number with zero real and imaginary
+   * components.
+   */
+  complex() : base_t() {}
+
+  /**
+   * Destroy this complex number.
+   */
+  ~complex() {}
+
+  /**
+   * Assign the specified value to the real part of this complex
+   * number and set imaginary part to zero.
+   *
+   * @tparam V type of value
+   * @param[in] x value to assign
+   * @return this complex number after assigning other
+   */
+  template <typename V>
+  complex_type& operator=(const V& x) {
+    return base_t::operator=(x);
+  }
```

Comment:
`V` must be assignable to the value type of the complex number.  It won't compile otherwise.  Adding an `enable_if` restriction would just cause the compiler to fail in a different way.  We don't need the restriction for disambiguation.

My understanding is that we generally do not want to try to explicitly enumerate all the possible instantiations of a template parameter with type traits.  We don't do that on any of the `prim` functions in our math library, for example.  You can try to pass a `std::string` to `log1m`, but it won't compile.